### DISCUSSION TITLE
ntfy integration with auth?

### DISCUSSION THREAD

**Discussion by Schachte** [juftin/camply#270] [Q&A] [answered]
I have ntfy running behind a reverse proxy and I want to enable basic auth on it. 

I noticed that using custom domains with ntfy doesn't seem supported, but I was able to grab the forwarding URL to ntfy.sh from the logs, which is just the md5 hash. I have 2 questions around this:

- Does camply support custom domains for ntfy? (Do you need help adding this?)
- Does camply support ntfy push notifications for ntfy self-hosted servers behind auth? (Do you need help adding this?)

**Comment by juftin** [comment, marked as answer]
It would be very easy to make `NTFY_ENDPOINT` a camply configuration entry, this should unlock you to use self-hosted ntfy servers https://github.com/juftin/camply/blob/5ed2359b5e58aa292b82150ee559fd0fa74df452/camply/config/notification_config.py#L68-L74

As far as auth goes, is that just an additional header that needs to be passed? All of the `ntfy` magic lies in this simple `requests` post  https://github.com/juftin/camply/blob/5ed2359b5e58aa292b82150ee559fd0fa74df452/camply/notifications/ntfy.py#L32-L60

If you're willing to help with a new PR, that would be great. Let me know - if so the original `NTFY` PR will be a useful reference:  https://github.com/juftin/camply/pull/248

**Reply by Schachte** [reply]
Makes sense, thanks!
